### PR TITLE
added batch normalization between layers, changed default weight initialization

### PR DESCRIPTION
**Pull request by iskandr**
Difference between initialization algorithms:
- "lecun_uniform" : divides weights by fan_in of layer 
- "glorot_uniform" divides by fan_in + fan_out

Another "small" change that could have significant impact on trained models: switched the decay rate of weights on imputed data back to "exponential" from "quadratic" (imputed data gets ignored after fewer epochs, models train faster).

### CODE REVIEW

**Comment by timodonnell** [comment]
LGTM
